apps/aot/src/utils/time-utils.ts

```diff
@@ -22,3 +22,15 @@ export function isValidAdventDay(selectedDay: number) {
   const currentAdventDay = getCurrentAdventDay();
   return selectedDay <= currentAdventDay;
 }
+
+export const getNextAdventDay = () => {
+  const currentDate = new Date();
+  const releaseTime = new Date(currentDate);
+  releaseTime.setUTCHours(5, 0, 0, 0);
+
+  if (releaseTime.getTime() <= currentDate.getTime()) {
```

Comment:
can you put this in english. getting confused haha

Reply:
i ran this at 3:13 PT and it gave me back 10pm PT. should be be 9pm PT? Since we release at 12am ET

![CleanShot 2024-10-29 at 15 14 58](https://github.com/user-attachments/assets/a21ab5d3-b5d7-4232-aaa4-eb3a8f9e028c)

Reply:
It's 4am rn because of daylight time, but in December 1 it's back to EST, so it should be 5AM. Yes, I can provide some documentation here to make code more clear.

It's basically just getting the next release date/time.

Reply:
So if yo want to test this in console for October, change to 4AM EDT:

```  
const releaseTime = new Date(currentDate);
 releaseTime.setUTCHours(4, 0, 0, 0);
```

Reply:
I can add this as doc:

```
This function calculates when the next 5 AM UTC release will be (12:00 AM EST) - either today's 5 AM if it hasn't happened yet, or tomorrow's 5 AM if today's has already passed. 
```

Reply:
To explain the `if (releaseTime.getTime() <= currentDate.getTime()) {` a bit.

1. Set releaseTime to current date, then set hour to be 5AM.
2. If we've already passed 5AM today, meaning we're at [5AM UTC - 23:59PM UTC], then we have to set release date to tomorrow's 5AM instead.
3. If we haven't passed 5AM today, meaning we're at [12:00AM UTC - 04:59AM UTC], then we we just keep it as 5AM of today since we haven't passed it yet.
4. Return .getTime()

Does that help explain?